lib/vagrant-proxyconf/config/env_proxy.rb

```diff
@@ -46,12 +49,22 @@ def config_for(key, value)
             # still in v1.0.x.
             value = value.inspect if key.name == :no_proxy
 
+            # Quoting `autoconfig` value as its a url
+            value = value.inspect if key.name == :autoconfig
+
             [var.upcase, var.downcase].map { |v| "export #{v}=#{value}\n" }.join
           end
         end
 
         def env_variable_name(key)
-          key.name == :no_proxy ? "no_proxy" : "#{key.name}_proxy"
+          #key.name == :no_proxy ? "no_proxy" : "#{key.name}_proxy"
+          if key.name == :no_proxy
+            return "no_proxy"
```

Comment:
As the `if` is the last statement of the method, the `return` can be dropped.

I would also use a `case` statatement here:

```ruby
case key.name
when :no_proxy
  "no_proxy"
when :autoconfig
  "auto_config_url"
else
  "#{key.name}_proxy"
end
```